errors/base-error.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const ExtendableError = require('es6-error')
+const isObject = require('101/is-object')
+
+/**
+ * Extendable base class that includes additional information that effects the
+ * way errors are reported to rollbar.
+ * @class
+ * @author Ryan Sandor Richards
+ * @extends es6-error~ExtendableError
+ */
+module.exports = class BaseError extends ExtendableError {
+  /**
+   * Creates a new base error.
+   * @param {String} message Message for the error.
+   * @param {Object} data Additional data for the error to be given to rollbar.
+   * @param {Object} reporting Reporting options.
```

Comment:
The correct way to write the JSDoc for this is with the parameters of the object. http://usejsdoc.org/tags-param.html. I feel this is pretty important in this case because this is not an arbitrary object like `data`.